lighthouse-core/audits/legacy-javascript.js

```diff
@@ -166,15 +168,13 @@ class LegacyJavascript extends Audit {
 
   static getPolyfillData() {
     return [
-      /* eslint-disable max-len */
       ['Array.prototype.fill', 'es6.array.fill'],
       ['Array.prototype.filter', 'es6.array.filter'],
       ['Array.prototype.find', 'es6.array.find'],
       ['Array.prototype.findIndex', 'es6.array.find-index'],
       ['Array.prototype.forEach', 'es6.array.for-each'],
       ['Array.from', 'es6.array.from'],
       ['Array.isArray', 'es6.array.is-array'],
-      ['Array.prototype.lastIndexOf', 'es6.array.last-index-of'],
```

Comment:
what we should detect is simply any polyfill that will never be in a bundle w/ `esmodules: true, bugfixes: true` (using either core-js@2 or core-js@3). everything being removed fails that criteria. if we add any back, it'll be because something changed upstream (and we can just add back to this list) or we accompany it with documentation for how and why to exclude (what we may do for typed-array, promise).

This list is derived from the diff of the debug output when you toggle `esmodule: true/false`. It may be good to print out that output in `legacy-javascript/run.js` (doing much more than that would be more work than I'd like)

___

sidenote on the line you commented on... last-index-of used to be set to an older version of safari but was bumped here: https://github.com/zloirock/core-js/commit/e4cea558aba8727a4f0a7051304755dac30fcc5e

no idea what bug this is preventing lol https://github.com/zloirock/core-js/blob/master/packages/core-js/internals/array-method-uses-to-length.js

Reply:
alright wfm

I'd think future me would be annoyed by having to hunt down an ancient PR and copy paste individual lines from a diff that no longer merges for this info but I'll defer to your future self since it's almost definitely going to be you anyway :)